.github/workflows/posix.yml

```diff
@@ -241,15 +265,43 @@ jobs:
       run: |
         tar -cf build/posix.tar -C build/ posix/
         rm -rf build/posix/
-    - name: Codecov
-      uses: codecov/codecov-action@v1
+    - name: Generate Coverage
+      run: |
+        ./script/test generate_coverage gcc
+    - uses: actions/upload-artifact@v2
       with:
-        fail_ci_if_error: true
+        name: cov-posix-pty-daemon-1
+        path: tmp/coverage.info
     - name: Keep POSIX Only
       run: |
         rm -rf build/x86_64-unknown-linux-gnu/
         tar -xf build/posix.tar -C build/
-    - name: Codecov
-      uses: codecov/codecov-action@v1
+    - name: Generate Coverage
+      run: |
+        ./script/test generate_coverage gcc
+    - uses: actions/upload-artifact@v2
+      with:
+        name: cov-posix-pty-daemon-2
+        path: tmp/coverage.info
+
+  upload-coverage:
+    needs:
+    - posix-cli
+    - posix-ncp
+    - posix-ncp-rcp-migrate
+    - posix-pty
+    - posix-pty-daemon
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Bootstrap
+      run: |
+        sudo apt-get --no-install-recommends install -y lcov
+    - uses: actions/download-artifact@v2
       with:
-        fail_ci_if_error: true
+        path: coverage/
+    - name: Upload Coverage
+      run: |
+        ls -R coverage/
+        lcov $(for i in $(find coverage/ -type f -name '*.info'); do echo -a $i; done) -o final.info
```

Comment:
The command is complex and used many times. Can we encapsulate the logic in `script/test` using a new command?

Reply:
Done.